### DISCUSSION TITLE
SlateDB would like to join Commonhaus

### DISCUSSION THREAD

**Discussion by criccomini** [commonhaus/foundation#213] [Joining Commonhaus]
### Project information

- **Project name**: SlateDB
- **Project website**: https://slatedb.io
- **Code repository**: https://github.com/slatedb/slatedb
- **License**: Apache 2.0

### Do you have authority to represent this project?

- [X] I own or maintain this project

### Describe your project

[SlateDB](https://slatedb.io/) is an embedded storage engine built as a [log-structured merge-tree](https://en.wikipedia.org/wiki/Log-structured_merge-tree) written in Rust. Unlike traditional LSM-tree storage engines, SlateDB writes data to object storage (S3, GCS, ABS, MinIO, Tigris, and so on). Leveraging object storage allows SlateDB to provide bottomless storage capacity, high durability, and easy replication.

SlateDB is useful for stream processing state management, durable execution persistence, durable caches, workflow orchestration, data lakes, and more. It's a library that's intended to be a building block for infrastructure that wishes to store row-based data on the cloud.

The project has committers from Responsive, Microsoft, Jamsocket, Databend, and Materialized View. It's currently being deployed at Tensorlake and Responsive.

### Motivation for joining Commonhaus

> What are you hoping to get out of joining Commonhaus?

We have three main goals:

- Donate IP and trademarks to a foundation
- Avoid bureaucracy and have the freedom to run our project in a way that suits us
- Get advice on governance structure

> Why Commonhaus as opposed to other organizations?

I have extensive experience with Apache (I'm on the PMC, created Apache Samza, and mentored Apache Airflow), but opted to avoid it for this project because it's a bit heavy weight. I was hoping CNCF might fit our needs. We applied to the sandbox. But after several months, I'm finding it very heavy on bureaucracy as well. It's a distraction, and getting in the way. @gunnarmorling turned me on to Commonhaus and it looks like a perfect fit for our needs.

> Any questions or concerns about joining Commonhaus?

Nope! Just excited to join. :)

### Things that must be done (bootstrap phase)

- [x] **Review the bylaws**, especially sections about [projects](https://www.commonhaus.org/bylaws/cf-membership.html#projects), [decision making](https://www.commonhaus.org/bylaws/decision-making.html), and [voting for significant decisions](https://www.commonhaus.org/bylaws/cf-council.html#voting).
- [x] **Read the [bootstrapping agreement](https://github.com/commonhaus/foundation/blob/main/agreements/bootstrapping/bootstrapping.md)**.
- [x] If all is well, and you agree to participate in the bootstrapping phase to help us build this thing, then **create a PR** with the following:

    1. A filled out copy of the bootstrapping agreement. You can make a copy of [bootstrapping.md](https://github.com/commonhaus/foundation/blob/main/agreements/bootstrapping/bootstrapping.md) file and check that in, or checkin a filled out [pdf](https://github.com/commonhaus/foundation/releases/download/final-draft/bootstrapping.pdf). Please follow naming conventions.
    2. An updated [CONTACTS.yaml](https://github.com/commonhaus/foundation/blob/main/CONTACTS.yaml) that names your project representative in the EGC section
    3. An updated [PROJECTS.yaml](https://github.com/commonhaus/foundation/blob/main/PROJECTS.yaml) that provides details / describes your project
    4. An updated [TRADEMARKS.md](https://github.com/commonhaus/foundation/blob/main/TRADEMARKS.md) that adds your project name and project logo (if you have one). Please include the footnote reference, as we are using your project name and logo with your permission until we have a formal agreement in place and resources are transferred.

    The commit should be "Signed-off-by" your GH login (`git commit -s -m "<project> bootstrapping agreement, yyyy-mm-dd"`)

**Comment by criccomini** [comment]
[SlateDB bootstrapping agreement, 2024-11-05](https://github.com/commonhaus/foundation/pull/212)

**Comment by gavinking** [comment]
This looks very interesting.

**Reply by gunnarmorling** [reply]
Woot, super cool!

**Comment by criccomini** [comment]
@gavinking One thing I should have asked in the PR: I'm not totally clear on what next steps are after opening the discussion and bootstrapping PR.

**Reply by gavinking** [reply]
@ebullient just started a discussion where the proposal gets voted on. I believe it's non-public. (There's currently one vote in favor.)

**Reply by criccomini** [reply]
I couldn't quite determine from https://www.commonhaus.org/bylaws/cf-council.html#voting how long votes typically take? Is it just until simple majority is reached? Trying to get a sense on time. =)

**Reply by ebullient** [reply]
Be patient. Folks are quiet, and we're still in a bootstrapping phase, so things move slowly.
This is a clearer cut case than some.

**Comment by ebullient** [comment]
🎉 Welcome to Commonhaus!

**Reply by criccomini** [reply]
Thank you! I am so excited to hear this. I've been singing the foundation's praises a lot lately. Wonderful news.

**Reply by ebullient** [reply]
=)

You have an invitation to join the organization pending. Accept that, and we'll get things moving.

**Reply by gavinking** [reply]
Welcome!!

**Reply by criccomini** [reply]
@ebullient Joined!

Also, random aside: it appears the SlateDB logo isn't appearing on commonhaus.org

**Reply by ebullient** [reply]
I didn't catch the svg path. You need to append ?raw=true to GH image links. I suppose I should update my tool to catch that..
 🙄
